src/compressed_tensors/utils/offload.py

```diff
@@ -49,68 +90,252 @@ def get_execution_device(module: Module) -> torch.device:
+    data = data.to(param.dtype)
+
+    # copy data into onloaded parameter if applicable
+    if param.device != "meta":
+        param.data.copy_(data)
+
+    # update offload dict
+    if has_offloaded_params(module):
+        weights_map = module._hf_hook.weights_map
+
+        # for upstreaming, better to add write capabilities to weight map classes first
+        if isinstance(weights_map, PrefixedDataset):
+            dataset = getattr(weights_map, "dataset", None)
+            if dataset is not None:
+                prefix = module._hf_hook.weights_map.prefix
+                key = f"{prefix}{name}"
+
+                offload_device = (
+                    dataset[key].device
+                    if key in dataset
+                    else next(iter(dataset.values())).device
+                )
+
+                if isinstance(dataset, OffloadedWeightsLoader):
+                    dataset.state_dict[key] = data.to(device=offload_device)
+                else:
+                    dataset[key] = data.to(device=offload_device)
+
+        elif isinstance(weights_map, dict):
+            offload_device = (
+                weights_map[name].device
+                if name in weights_map
+                else next(iter(weights_map.values())).device
+            )
+            weights_map[name] = data.to(device=offload_device)
+
+        elif isinstance(weights_map, OffloadedWeightsLoader):
+            raise NotImplementedError()
+
+        else:
+            raise NotImplementedError(
+                "Updating offload data not implemented for weights_map of type "
+                f"{type(weights_map)}"
+            )
+
+
+def delete_offload_parameter(module: torch.nn.Module, name: str):
+    """
+    Delete a parameter from a module which may be offloaded
+
+    :param module: maybe offloaded module
+    :param name: name of parameter being deleted
+    """
+    delattr(module, name)
+
+    if has_offloaded_params(module):
+        weights_map = module._hf_hook.weights_map
+
+        # for upstreaming, better to add write capabilities to weight map classes first
+        if isinstance(weights_map, PrefixedDataset):
+            dataset = weights_map.dataset
+            prefix = weights_map.prefix
+            if dataset is not None:
+                del dataset[f"{prefix}{name}"]
+
+        elif isinstance(weights_map, dict):
+            del weights_map[name]
+
+        elif isinstance(weights_map, OffloadedWeightsLoader):
+            raise NotImplementedError()
+
+        elif weights_map is not None:
+            raise NotImplementedError(
+                f"Cannot delete parameter from weights_map of type {type(weights_map)}"
+            )
+
+
+@check_accelerate(fallback=contextlib.nullcontext())
+@contextlib.contextmanager
+def disable_hf_hook(module: torch.nn.Module, recurse: bool = False):
+    hooks = {}
+
+    def collect_hooks(module):
+        nonlocal hooks
+        if hasattr(module, "_hf_hook"):
```

Comment:
Hm, not quite sure how this relates to `disable_hf_hook`. Newly initialized quantization parameters are added to the prefix dict by `register_offload_parameter`

Reply:
Previously it seems like after adding the scales/zp to the layers, we were updating the PreFix dict to add the quant parameters, removing the hooks, wrapping the forward pass and then re-adding the hooks. 

Now we're using `register_offload_parameter` so just trying to understand why we can skip the Prefix dict update step.

Reply:
The "prefix dict update step" is implemented [here](https://github.com/neuralmagic/compressed-tensors/pull/193/files#diff-8a76ba965bc57e883a4231f3d0258b6eb155a8a75b4644d11b9bc87926411fa2R200-R215), which is called by `register_offload_parameter`.

Since updating the prefix dict is covered by `register_offload_parameter`, we no longer have to do this explicitly.